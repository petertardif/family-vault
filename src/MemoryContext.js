import React from 'react';

export const MemoryContext = React.createContext({
  on: '', 
  deleteOn: '',
  toggle: () => {},
  deleteToggle: () => {},
  memories: [],
  familyMembers: [],
  addMemory: () => {},
  deleteMemory: () => {},
  addFamilyMember: () => {},
  deleteFamilyMember: () => {},
  updateMemoryList: () => {},
});