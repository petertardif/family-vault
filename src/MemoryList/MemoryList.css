.Memorylist__title {
  font-family: 'Baloo', cursive;
  font-size: 100px;
}