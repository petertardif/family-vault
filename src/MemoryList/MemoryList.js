import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import Memory from '../Memory/Memory';
import { MemoryContext } from '../MemoryContext';
import './MemoryList'
import '../Button/Button.css';

class MemoryList extends Component {
  static defaultProps = {
    memories: [],
    familyMembers: [],
    history: {
      push: () => { }
    }
  }

  static contextType = MemoryContext;
  
  componentWillMount() {
      fetch(`https://agile-fortress-94521.herokuapp.com/api/memories`)
      // fetch(`https://agile-fortress-94521.herokuapp.com/api/family-members`)
      
      .then((memoriesResponse) => {
        if(!memoriesResponse.ok) {
          return memoriesResponse.json().then(error => Promise.reject(error))
        }
        return memoriesResponse.json()
      })
      .then((memories) => {
        this.context.updateMemoryList(memories)
      })
      .catch(err => {
        this.setState({
          error: err.message
        });
      })
  }

  handleClick = () => {
    this.props.history.push('/userlanding');
  }

  render() {
    return (
      <section className='MemoryList'>
        <h2>Memories</h2> 
        <ul className='MemoryList_list' aria-live='polite'>
          {this.context.memories.map(memory =>
            <Memory
              key={memory.id}
              {...memory}
            />
          )}
        </ul>
        <section className='buttons'>
          <button onClick={this.handleClick} className='Button blue'>Back to home page</button>
        </section>
      </section>
    );
  }
}

export default withRouter(MemoryList);