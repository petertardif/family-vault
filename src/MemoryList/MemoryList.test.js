import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MemoryList from './MemoryList'
import MemoryContext from '../MemoryContext';

describe(`MemoryList component`, () => {
  const props = {
    memories: [
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "memory_title": "Dogs",
        "date_updated": "2019-01-03T00:00:00.000Z",
        "familymember_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "memory_desc": "Corporis accusamus placeat.\n \rUnde."
      },
      {
        "id": "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
        "memory_title": "Cats",
        "date_updated": "2018-08-15T23:00:00.000Z",
        "familymember_id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "memory_desc": "Eos\n \rlaudantium."
      },
      {
        "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "memory_title": "Pigs",
        "date_updated": "2018-03-01T00:00:00.000Z",
        "familymember_id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "memory_desc": "Occaecati dignissimos\nvoluptatum nihil."
      },
      {
        "id": "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
        "memory_title": "Birds",
        "date_updated": "2019-01-04T00:00:00.000Z",
        "familymember_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "memory_desc": "Eum culpa odit."
      },
    ]
  }

  it('renders a .MemoryList by default', () => {
    const wrapper = shallow(<MemoryList />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Memory in ul for each memory in array', () => {
    const ul = shallow(<MemoryList {...props} />)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})
