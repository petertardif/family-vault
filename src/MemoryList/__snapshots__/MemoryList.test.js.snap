// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemoryList component renders a .MemoryList by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`MemoryList component renders a Memory in ul for each memory in array 1`] = `null`;
